docs/sourcemap-api.asciidoc

```diff
@@ -0,0 +1,84 @@
+[[sourcemap-api]]
+== Sourcemap API
+
+NOTE: Sourcemap Support is an experimental feature.
+
+The APM Server exposes an API Endpoint to upload sourcemaps. 
+When <<sourcemap-endpoint,sourcemaps are uploaded>> and <<sourcemap-enable,frontend support is enabled>>, sourcemapping is automatically applied to all `errors` and `transactions` recorded with the https://github.com/elastic/apm-agent-js-base[experimental javascript frontend agent]. 
+
+
+[[sourcemap-endpoint]]
+[float]
+=== Endpoint
+To upload a sourcemap you need to send a `HTTP POST` request to the APM Server `sourcemaps` endpoint:
+
+[source,bash]
+------------------------------------------------------------
+http(s)://{hostname}:{port}/v1/client-side/sourcemaps
+------------------------------------------------------------
+
+
+[[sourcemap-request-fields]]
+[float]
+=== Request Fields
+The request must be sent using `Content-Type` `multipart/form-data`. 
+The request must contain following fields:
+
+* `service_name`
+* `service_version`
+* `bundle_filepath`
+* `sourcemap`: The sourcemap must be attached to the request as a `file` upload.
+The sourcemap must match the specification for 
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[Sourcemap Revision 3].
+
+
+The combination of the `service_name`, `service_version` and `bundle_filepath` is used to identify a `sourcemap`. See <<sourcemap-apply,applying sourcemaps>> for more information.  
+
+
+[[sourcemap-enable]]
+[float]
+=== Enable Frontend Support
+To try out experimental frontend support `apm-server.frontend.enabled` needs to be set to `true`.
+See https://github.com/elastic/apm-server/blob/{doc-branch}/apm-server.reference.yml[`apm-server.reference.yml`] for more configuration options.
+
+
+[[sourecmap-api-examples]]
+[float]
+=== Examples
```

Comment:
I'd prefer the example to appear closer to the section where we describe how to upload the source map.